src/compiler/compile/render_ssr/handlers/Text.ts

```diff
@@ -14,5 +14,12 @@ export default function(node: Text, renderer: Renderer, _options: RenderOptions)
 		text = escape_html(text);
 	}
 
+	// https://html.spec.whatwg.org/multipage/syntax.html#element-restrictions
+	// when parsing HTML, if <pre> or <textarea> starts with `\n`, it is ignored
+	// add another `\n` for it to be applied
+	if (text[0] === '\n' && node.parent && node.parent.type === 'Element' && ((node.parent as Element).name === 'pre' || (node.parent as Element).name === 'textarea') && node === node.parent.children[0]) {
```

Comment:
This sounds related to #7264, but this reads like this would introduce the (according the spec) wrong behavior in SSR, too. I feel like we should fix the DOM-render (#7280 is an attempt to do it) in the long run - I'm okay with this change if it means we align SSR and DOM behavior.

Reply:
i think https://github.com/sveltejs/svelte/pull/7280 is a better fix, which aligns the svelte `<pre>` tag parsing with the browser's html parsing.

i'll wait for https://github.com/sveltejs/svelte/pull/7280 land and revert my change.

Reply:
#7280 is merged now